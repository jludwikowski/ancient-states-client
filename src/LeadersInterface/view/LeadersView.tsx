import React from "react"
import {LeadersList} from "../components/LeadersList";
import Leader from "../../Models/Leader";

export class LeadersView extends React.Component {

    leadersList: Leader[] = [];

    componentDidMount(): void {

    };

    onDismiss(id:number) {

    };

    render() {
        return(
            <>
                <LeadersList leadersList={this.leadersList} onDismiss={this.onDismiss} />
            </>
        )}
}
