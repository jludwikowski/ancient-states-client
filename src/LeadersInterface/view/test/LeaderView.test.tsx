import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import { LeadersView } from '../LeadersView';

describe('LeadersView', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<LeadersView />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Should match snapshot', () => {
        const component = shallow(<LeadersView/>);
        expect(component).toMatchSnapshot();
    });
});
