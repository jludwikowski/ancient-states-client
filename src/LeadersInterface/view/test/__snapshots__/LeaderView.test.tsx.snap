// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeadersView Should match snapshot 1`] = `
<Fragment>
  <LeadersList
    leadersList={Array []}
    onDismiss={[Function]}
  />
</Fragment>
`;
